src/config/util/tls.rs

```diff
@@ -0,0 +1,71 @@
+use anyhow::{Context, Error, Result};
+use rustls::internal::pemfile;
+use rustls::{Certificate, PrivateKey};
+use serde::Deserialize;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
+use std::str::FromStr;
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub enum PrivateKeyAlgorithm {
+    #[serde(rename = "rsa")]
+    Rsa,
+    #[serde(rename = "pkcs8")]
+    Pkcs8,
+}
+
+impl FromStr for PrivateKeyAlgorithm {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "rsa" => Ok(PrivateKeyAlgorithm::Rsa),
+            "pkcs8" => Ok(PrivateKeyAlgorithm::Pkcs8),
+            _ => Err(Error::msg(format!(
+                "Invalid algorithm name provided for TLS key. {}",
+                s
+            ))),
```

Comment:
I think this can be improved slightly with Anyhow's [`bail!`](https://docs.rs/anyhow/1/anyhow/macro.bail.html) macro:

```suggestion
            _ => anyhow::bail!(
                "Invalid algorithm name provided for TLS key. {}",
                s
            ),
```